forms-flow-web/src/components/Form/Item/Submission/Item/Edit.js

```diff
@@ -0,0 +1,168 @@
+import React, {useEffect} from 'react';
+import {connect, useDispatch, useSelector} from 'react-redux'
+import { selectRoot, resetSubmissions, saveSubmission, Form, selectError, Errors } from 'react-formio';
+import { push } from 'connected-react-router';
+
+import Loading from '../../../../../containers/Loading'
+
+import {setFormSubmissionError, setFormSubmissionLoading} from '../../../../../actions/formActions';
+import SubmissionError from '../../../../../containers/SubmissionError';
+import {getUserRolePermission} from "../../../../../helper/user";
+import {CLIENT} from "../../../../../constants/constants";
+import {
+  CLIENT_EDIT_STATUS,
+  UPDATE_EVENT_STATUS,
+  getProcessDataReq
+} from "../../../../../constants/applicationConstants";
+import {useParams} from "react-router-dom";
+import {updateApplicationEvent} from "../../../../../apiManager/services/applicationServices";
+import LoadingOverlay from "react-loading-overlay";
+import {toast} from "react-toastify";
+const Edit = React.memo((props) => {
+  const dispatch = useDispatch();
+  const {formId, submissionId} = useParams();
+  const {
+    hideComponents,
+    onSubmit,
+    options,
+    errors,
+    onFormSubmit,
+    onCustomEvent,
+    task,
+    user,
+    form: { form, isActive: isFormActive },
+    submission: { submission, isActive: isSubActive, url },
+  } = props;
+  
+  const applicationStatus = useSelector(state => state.applications.applicationDetail?.applicationStatus || '');
+  const userRoles = useSelector((state) => {
+    return selectRoot("user", state).roles;
+  });
+  const applicationDetail = useSelector(state=>state.applications.applicationDetail);
+  const isFormSubmissionLoading = useSelector(state=>state.formDelete.isFormSubmissionLoading);
+  useEffect(() => {
+    if (applicationStatus && !onFormSubmit) {
+      if (getUserRolePermission(userRoles, CLIENT) && !CLIENT_EDIT_STATUS.includes(applicationStatus)) {
+        dispatch(push(`/form/${formId}/submission/${submissionId}`));
+      }
+    }
+  }, [applicationStatus, userRoles, dispatch, submissionId, formId, onFormSubmit ]);
+
+  if ((isFormActive ||  (isSubActive && !isFormSubmissionLoading))) {
+      return <Loading />;
+  }
+
+  // Pass along the current task with the given submission
+  // so it can be used for validation purposes.
+  const submissionWithTask = _.merge({}, {
+        data: {
+            task: {
+                assignedToMe: task?.assignee && user?.preffered_username === task?.assignee,
+                ...(task || {})
+            }
+        }
+    }, submission);
```

Comment:
Added the above

Reply:
This merge function is very interesting!